editor/editor.tsx

```diff
@@ -416,20 +440,37 @@ function Editor(props: any) {
 
             let editedGos;
             let valid;
-            try {
-                editedGos = JSON.parse(stripJsonComments(code));
-                valid = gosling.validateGoslingSpec(editedGos);
-                setLog(valid);
-            } catch (e) {
-                const message = '✘ Cannnot parse the code.';
-                console.warn(message);
-                setLog({ message, state: 'error' });
+
+            if (language === 'json') {
+                try {
+                    editedGos = JSON.parse(stripJsonComments(code));
+                    valid = gosling.validateGoslingSpec(editedGos);
+                    setLog(valid);
+                } catch (e) {
+                    const message = '✘ Cannnot parse the code.';
+                    console.warn(message);
+                    setLog({ message, state: 'error' });
+                }
+            } else if (language === 'javascript') {
+                try {
+                    const transplieCode = ts.transpile(jsCode);
+                    editedGos = window.Function(`${transplieCode}\n return spec`)();
```

Comment:
Seems like a typo (`transp_li_eCode`)?

```suggestion
                    const transpiledCode = ts.transpile(jsCode);
                    editedGos = window.Function(`${transpiledCode}\n return spec`)();
```